linera-rpc/src/grpc/client.rs

```diff
@@ -434,6 +437,26 @@ impl ValidatorNode for GrpcClient {
         Ok(certs_collected)
     }
 
+    #[instrument(target = "grpc_client", skip(self), err(level = Level::WARN), fields(address = self.address))]
+    async fn download_certificates_by_heights(
+        &self,
+        chain_id: ChainId,
+        heights: Vec<BlockHeight>,
+    ) -> Result<Vec<ConfirmedBlockCertificate>, NodeError> {
```

Comment:
Do we need the same `is_empty` check here, too?

Reply:
Done in [Add loop and early exit to download_certs_at_heights](https://github.com/linera-io/linera-protocol/pull/4373/commits/19e2145238a9094d1819a9fb3777d5ba00106573)